tests/test_controller/test_action_normalization.py

```diff
@@ -0,0 +1,174 @@
+    denormalized = controller.action_normalizer.denormalize(normalized)
+
+    np.testing.assert_array_almost_equal(actions, normalized)
+    np.testing.assert_array_almost_equal(actions, denormalized)
+
+
+def test_min_max_normalizer_behavior() -> None:
+    """Test that min_max normalizer correctly scales values to [-1, 1] range."""
+    task_config = CylinderPushConfig()
+    task = CylinderPush()
+    optimizer_config = CrossEntropyMethodConfig()
+    optimizer = CrossEntropyMethod(optimizer_config, task.nu)
+
+    controller = Controller(
+        ControllerConfig(action_normalizer="min_max"),
+        task,
+        task_config,
+        optimizer,
+        optimizer_config,
+        rollout_backend="mujoco",
+    )
+
+    # Check that the normalizer is initialized as min_max
+    assert isinstance(controller.action_normalizer, MinMaxNormalizer)
+
+    # Test with actions at the bounds
+    min_actions = task.actuator_ctrlrange[:, 0]
+    max_actions = task.actuator_ctrlrange[:, 1]
+
+    # Normalize min and max values
+    min_normalized = controller.action_normalizer.normalize(min_actions)
+    max_normalized = controller.action_normalizer.normalize(max_actions)
+
+    # Should be close to -1 and 1 respectively
+    np.testing.assert_array_almost_equal(min_normalized, -np.ones_like(min_normalized))
+    np.testing.assert_array_almost_equal(max_normalized, np.ones_like(max_normalized))
+
+    # Test round-trip normalization
+    actions = np.random.uniform(min_actions, max_actions, (10, task.nu))
+    normalized = controller.action_normalizer.normalize(actions)
+    denormalized = controller.action_normalizer.denormalize(normalized)
+
+    np.testing.assert_array_almost_equal(actions, denormalized)
+
+    # Check that normalized values are in [-1, 1] range
+    assert np.all(normalized >= -1.0 - 1e-6)
+    assert np.all(normalized <= 1.0 + 1e-6)
+
+
+def test_running_mean_std_normalizer() -> None:
+    """Test that running_mean_std normalizer correctly updates and normalizes."""
+    task_config = CylinderPushConfig()
+    task = CylinderPush()
+    optimizer_config = CrossEntropyMethodConfig()
+    optimizer = CrossEntropyMethod(optimizer_config, task.nu)
+
+    controller = Controller(
+        ControllerConfig(action_normalizer="running_mean_std"),
+        task,
+        task_config,
+        optimizer,
+        optimizer_config,
+        rollout_backend="mujoco",
+    )
+
+    # Check that the normalizer is initialized as running_mean_std
+    assert isinstance(controller.action_normalizer, RunningMeanStdNormalizer)
+
+    # Test with some random actions
+    actions = np.random.randn(10, task.nu)
+    controller.action_normalizer.update(actions)
+
+    # Check that statistics were updated
+    assert controller.action_normalizer.count == 10
+    assert not np.allclose(controller.action_normalizer.mean, np.zeros(task.nu))
+    assert not np.allclose(controller.action_normalizer.std, np.ones(task.nu))
+
+    # Test normalization
```

Comment:
you should not only test that the inverse is roughly the identity, you should test the actual statistics inside the normalizer